### PR TITLE
Fix mbed-tool support

### LINKED ISSUE #75: CMake build fails with `Unknown CMake command "mbed_set_mbed_target_linker_script"`
### Description of defect

```
#### STDERR ####

#### End of STDOUT/STDERR ####

IMAGE is /builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@3/examples/mbed-os-example-sockets/cmake_build/FVP_MPS2_M3/develop/GCC_ARM/mbed-os-example-sockets.bin
[EXAMPLES]> INFO     In folder 'mbed-os-example-cellular'
[EXAMPLES]> INFO     In folder 'mbed-os-example-lorawan'
[EXAMPLES]> INFO     In folder 'mbed-os-example-devicekey'
[EXAMPLES]> INFO     In folder 'mbed-os-example-thread-statistics'
[EXAMPLES]> INFO     Compiling mbed-os-example-thread-statistics FVP_MPS2_M3 GCC_ARM
[EXAMPLES]> INFO     Executing command 'mbed-tools compile -t GCC_ARM -m FVP_MPS2_M3 -c'...

#### STDOUT ####
Configuring project and generating build system...
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.7 (found version "3.7.4") found components: Interpreter 
-- Checking for Python package prettytable -- found
-- Checking for Python package future -- found
-- Checking for Python package jinja2 -- found
-- Checking for Python package intelhex -- found
-- Configuring incomplete, errors occurred!
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@3/examples/mbed-os-example-thread-statistics/cmake_build/FVP_MPS2_M3/develop/GCC_ARM/CMakeFiles/CMakeOutput.log".
See also "/builds/workspace/mbed-os-ci_cmake-example-GCC_ARM-dev@3/examples/mbed-os-example-thread-statistics/cmake_build/FVP_MPS2_M3/develop/GCC_ARM/CMakeFiles/CMakeError.log".

#### STDERR ####
CMake Error at CMakeLists.txt:18 (mbed_set_mbed_target_linker_script):
  Unknown CMake command "mbed_set_mbed_target_linker_script".


ERROR: CMake invocation failed!
```

#### Target(s) affected by this defect ?
FVP_MPS2_M3

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM

#### What version of Mbed-os are you using (tag or sha) ?

9bbc8c914c7e 

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
Mbed CLI 2

#### How is this defect reproduced ?
`mbed-tools compile -t GCC_ARM -m FVP_MPS2_M3 -c`

### PR DESCRIPTION
**Pull request by harmut01**
Unfortunately the original PR (#76) was merged into master.
 
Fixes #75

`.travis.yml`:
* Update mbed-tools commands

`README.md`:
* Add CLI2 support documentation.

`CMakeLists.txt`:
* Remove mbed_set_mbed_target_linker_script() call which is no
  longer present in Mbed OS.
* Remove deprecated command mbed_configure_app_target

### CODE REVIEW

**Review by LDong-Arm — APPROVED** [review]

**Review by wernerlewis — APPROVED** [review]

**Review by Patater — CHANGES_REQUESTED** [review]
Needs to pass Travis CI before we can merge

**Review by Patater — APPROVED** [review]
Should be good to go now, pending Travis CI. Thanks!

**Pull request merged**
